org.eclipse.lsp4j.debug/src/main/java/org/eclipse/lsp4j/debug/adapters/StartDebuggingRequestArgumentsConfigurationTypeAdapter.java

```diff
@@ -0,0 +1,42 @@
+/******************************************************************************
+ * Copyright (c) 2022 KamasamaK and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ ******************************************************************************/
+
+package org.eclipse.lsp4j.debug.adapters;
+
+import java.util.function.Predicate;
+
+import org.eclipse.lsp4j.debug.AttachRequestArguments;
+import org.eclipse.lsp4j.debug.LaunchRequestArguments;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EitherTypeAdapter;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EitherTypeAdapter.PropertyChecker;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+
+public class StartDebuggingRequestArgumentsConfigurationTypeAdapter implements TypeAdapterFactory {
+	private static final TypeToken<Either<LaunchRequestArguments, AttachRequestArguments>> ELEMENT_TYPE = new TypeToken<Either<LaunchRequestArguments, AttachRequestArguments>>() {
+	};
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+		Predicate<JsonElement> leftChecker = new PropertyChecker("noDebug");
```

Comment:
Reading this makes it look to me like something is missing from the specification. There is no way to distinguish `launch` and `attach` on the receiving end of this. In https://github.com/microsoft/debug-adapter-protocol/issues/79#issuecomment-1230242253 there is a `request` key that is either `launch` or `attach`, but that isn't explicit in the final spec.

As a result, what is implemented here, basically always creates the `AttachRequestArguments` (because `noDebug` is most likely not present), even if there was a `"request": "launch"` in the JSON. 

I have raised https://github.com/microsoft/debug-adapter-protocol/issues/333

Reply:
I noticed that as well, so I didn't bother adding a type adapter for `RestartArguments`, which has the same type. But I guess that was a problem since #618 added it, so I just copied that implementation.

Reply:
I see. I guess we'll have to revisit that sometime.

Reply:
Removed this file